### PR TITLE
feat(serve-static): add serve-static-middleware

### PR DESCRIPTION
**Pull request by zdravko-georgiev**
This PR defines a `serve-static-middleware` for serving local files using [serve-static](https://www.npmjs.com/package/serve-static)

### CODE REVIEW

File: packages/serve-static-middleware/src/base/utils.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import type { ServeStaticOptions } from 'serve-static';
+import { relative, isAbsolute, join } from 'path';
+
+/**
+ * Resolves the serve static options from a configuration object.
+ *
+ * @param options configuration object
+ * @returns resolved serve static options
+ */
+export const resolveServeStaticOptions = (options: ServeStaticConfig | PathConfig): ServeStaticOptions => {
```
Comment (tobiasqueck) [inline packages/serve-static-middleware/src/base/utils.ts:11]: Could you please add to the function description why we are doing the filtering

File: packages/serve-static-middleware/test/base/servestatic.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { serveStaticMiddleware } from '../../src';
+import type { ServeStaticConfig } from '../../src';
+import { NullTransport, ToolsLogger } from '@sap-ux/logger';
+import * as expressServeStatic from 'serve-static';
+import { relative, join } from 'path';
+
+jest.mock('serve-static', () => ({
```
Comment (tobiasqueck) [inline packages/serve-static-middleware/test/base/servestatic.test.ts:7]: Are both line 7 and 16 required? Looks like both have the same intention

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
* please change the changeset to minor
* also add express as peerDependency (since you use `Router()`)
* `README.md` missing

File: packages/serve-static-middleware/src/base/utils.ts (hunk 1 above)
Comment (zdravko-georgiev) [reply packages/serve-static-middleware/src/base/utils.ts:13]: Sure, done with https://github.com/SAP/open-ux-tools/pull/1673/commits/1532eb52f836078b3adeaaab0faeaafa0884e907

**Review by tobiasqueck — APPROVED** [review]
- changeset ok
- code is straight forward
- good test coverage
- readme explains usage well
- local test successful

**Review by Klaus-Keller — APPROVED** [review]
Thanks @zdravko-georgiev!
- code is easy to understand and well commented
- test coverage is great
- did a local test, no issues found

Approved from my side.

**Comment by zdravko-georgiev** [comment]
Thanks!

**Pull request merged**